src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] list;
+    private int size;
+
+    public ArrayList() {
+        list = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (checkCapacity()) {
+            resize();
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (checkCapacity()) {
+            resize();
+        }
+        System.arraycopy(list, index, list, index + 1, size - index);
+        list[index] = value;
+        size++;
     }
 
     @Override
-    public void addAll(List<T> list) {
-
+    public void set(T value, int index) {
+        checkIndexBounds(index);
+        list[index] = value;
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
-    public void set(T value, int index) {
-
+    public T get(int index) {
+        checkIndexBounds(index);
+        return (T) list[index];
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndexBounds(index);
+        T removed = (T) list[index];
+        trimArray(index);
+        list[size] = null;
```

Comment:
```suggestion
        list[--size] = null;
```
we should decrement the size, right?

Reply:
I can see that you are decrementing it in `trim()`
nice!